src/ucp/tag/rndv.c

```diff
@@ -348,19 +433,31 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_proto_progress_rndv_get_zcopy, (self),
         return UCS_INPROGRESS;
     }
 
-    rsc_index = ucp_ep_get_rsc_index(rndv_req->send.ep, rndv_req->send.lane);
+    lane_idx = rndv_req->send.rndv_get.lane_idx;
+    lane = ucp_tag_rndv_get_lane_info(rndv_req, lane_idx)->lane;
+    rkey = ucp_tag_rndv_rkey(rndv_req, lane_idx);
+
+    rndv_req->send.rndv_get.lane_idx = (lane_idx + 1) %
```

Comment:
maybe to separate in a special function? we may want to have more complex distribution in future.

Reply:
let's separate it after this complex distribution is defined :)